### DISCUSSION TITLE
How to create a multi-value text box

### DISCUSSION THREAD

**Discussion by amtopel** [microsoft/microsoft-ui-xaml#8351] [Ideas]
I was wondering if there's any way to create a text box, or some sort of equivalent control, that's designed for entering (and deleting) multiple discrete values.

One example of this is in e-mail applications, where you enter the recipient addresses.  Often as you enter the addresses, they may have some sort of visual separation.  They'll also start a new item after you type a semi-colon.  And they may even have a close box:

<img width="366" alt="image" src="https://user-images.githubusercontent.com/26663310/229325943-294c92f1-63d6-410a-ab3d-88950a09e38d.png">

Has anyone accomplished this using WinUI?

**Comment by marcelwgn** [comment]
While there is no "out of the box" control included in the WinAppSDK, the [Windows Community Toolkit](https://learn.microsoft.com/en-us/windows/communitytoolkit/) has the [TokenizingTextBox](https://learn.microsoft.com/en-us/windows/communitytoolkit/controls/tokenizingtextbox) control which sounds like exactly what you want to have. I'm not sure though how it works with regard to create a tag once the users enters `;`.

**Reply by amtopel** [reply]
Thank you very much.  Can that be used with WinUI 3 C++?  Or would that require XAML Islands?
